src/graphic/Fast3D/gfx_pc.cpp

```diff
@@ -2578,670 +2620,1067 @@ void gfx_set_framebuffer(int fb, float noise_scale);
+    cmd = *cmd0;
+    uls = C0(16, 16);
+    ult = C0(0, 16);
+    dsdx = C1(16, 16);
+    dtdy = C1(0, 16);
+    gfx_dp_texture_rectangle(ulx, uly, lrx, lry, tile, uls, ult, dsdx, dtdy, opcode == G_TEXRECTFLIP);
+    return false;
+}
+
+bool gfx_image_rect_handler_custom(Gfx** cmd0) {
+    Gfx* cmd = *cmd0;
+    int16_t tile, iw, ih;
+    int16_t x0, y0, s0, t0;
+    int16_t x1, y1, s1, t1;
+    tile = C0(0, 3);
+    iw = C1(16, 16);
+    ih = C1(0, 16);
+    cmd = ++(*cmd0);
+    x0 = C0(16, 16);
+    y0 = C0(0, 16);
+    s0 = C1(16, 16);
+    t0 = C1(0, 16);
+    cmd = ++(*cmd0);
+    x1 = C0(16, 16);
+    y1 = C0(0, 16);
+    s1 = C1(16, 16);
+    t1 = C1(0, 16);
+    gfx_dp_image_rectangle(tile, iw, ih, x0, y0, s0, t0, x1, y1, s1, t1);
+
+    return false;
+}
+
+bool gfx_fill_rect_handler_rdp(Gfx** cmd0) {
+    Gfx* cmd = *(cmd0);
+
+    gfx_dp_fill_rectangle(C1(12, 12), C1(0, 12), C0(12, 12), C0(0, 12));
+    return false;
+}
+
+bool gfx_fill_wide_rect_handler_custom(Gfx** cmd0) {
+    Gfx* cmd = *(cmd0);
+    int32_t lrx, lry, ulx, uly;
+
+    lrx = (int32_t)(C0(0, 24) << 8) >> 8;
+    lry = (int32_t)(C1(0, 24) << 8) >> 8;
+    cmd = ++(*cmd0);
+    ulx = (int32_t)(C0(0, 24) << 8) >> 8;
+    uly = (int32_t)(C1(0, 24) << 8) >> 8;
+    gfx_dp_fill_rectangle(ulx, uly, lrx, lry);
+
+    return false;
+}
+
+bool gfx_set_scissor_handler_rdp(Gfx** cmd0) {
+    Gfx* cmd = *(cmd0);
+
+    gfx_dp_set_scissor(C1(24, 2), C0(12, 12), C0(0, 12), C1(12, 12), C1(0, 12));
+    return false;
+}
+
+bool gfx_set_z_img_handler_rdp(Gfx** cmd0) {
+    Gfx* cmd = *(cmd0);
+
+    gfx_dp_set_z_image(seg_addr(cmd->words.w1));
+    return false;
+}
+
+bool gfx_set_c_img_handler_rdp(Gfx** cmd0) {
+    Gfx* cmd = *(cmd0);
+
+    gfx_dp_set_color_image(C0(21, 3), C0(19, 2), C0(0, 11), seg_addr(cmd->words.w1));
+    return false;
+}
+
+bool gfx_rdp_set_other_mode_rdp(Gfx** cmd0) {
+    Gfx* cmd = *(cmd0);
+
+    gfx_dp_set_other_mode(C0(0, 24), (uint32_t)cmd->words.w1);
+    return false;
+}
+
+bool gfx_bg_copy_handler_s2dex(Gfx** cmd0) {
+    Gfx* cmd = *(cmd0);
+
+    if (!markerOn) {
+        gfx_s2dex_bg_copy((uObjBg*)cmd->words.w1); // not seg_addr here it seems
+    }
+    return false;
+}
+
+bool gfx_bg_1cyc_handler_s2dex(Gfx** cmd0) {
+    Gfx* cmd = *(cmd0);
+
+    gfx_s2dex_bg_1cyc((uObjBg*)cmd->words.w1);
+    return false;
+}
+
+bool gfx_obj_rectangle_handler_s2dex(Gfx** cmd0) {
+    Gfx* cmd = *(cmd0);
+
+    if (!markerOn) {
+        gfx_s2dex_rect_copy((uObjSprite*)cmd->words.w1); // not seg_addr here it seems
+    }
+    return false;
+}
+
+bool gfx_extra_geometry_mode_handler_custom(Gfx** cmd0) {
+    Gfx* cmd = *(cmd0);
+
+    gfx_sp_extra_geometry_mode(~C0(0, 24), (uint32_t)cmd->words.w1);
+    return false;
+}
+
+bool gfx_stubbed_command_handler(Gfx** cmd0) {
+    return false;
+}
+
+bool gfx_spnoop_command_handler_f3dex2(Gfx** cmd0) {
+    return false;
+}
+
+const static std::unordered_map<int8_t, GfxOpcodeHandlerFunc> rdpHandlers = {
```

Comment:
instead of ifdefing in the functions we could have copies of the functions and then set up the map with ifdefs for the appropriate functions